apps/web/src/pages/templates/editor_v2/useWorkflowDetailPageForm.ts

```diff
@@ -30,34 +36,44 @@ export const useWorkflowDetailPageForm = ({ templateId, workflow }: UseWorkflowD
     },
   });
 
-  const hasChanges = Object.keys(formState.dirtyFields).length > 0;
+  const hasChanges = Object.keys(dirtyFields).length > 0;
 
-  const onSubmit: SubmitHandler<WorkflowDetailFormContext> = () => {
-    if (formState.dirtyFields?.preferences) {
-      updateWorkflowChannelPreferences(getValues('preferences'));
-    }
+  const onSubmit: SubmitHandler<WorkflowDetailFormContext> = async (data) => {
+    try {
+      if (dirtyFields?.preferences) {
+        updateWorkflowChannelPreferences(getValues('preferences'));
+      }
 
-    if (formState.dirtyFields?.general) {
-      const { workflowId, ...templateValues } = getValues('general');
-      updateTemplateMutation({ id: templateId, data: { ...templateValues, identifier: workflowId } });
-    }
+      if (dirtyFields?.general) {
+        const { workflowId, ...templateValues } = getValues('general');
+        await updateTemplateMutation({ id: templateId, data: { ...templateValues, identifier: workflowId } });
+      }
 
-    reset();
+      successMessage('Workflow updated successfully');
+
+      reset(data);
```

Comment:
reset with the form data, otherwise it sets to empty strings(the default values)